test-poc/framework/src/main/java/org/keycloak/test/framework/config/KeycloakSmallryeConfig.java

```diff
@@ -0,0 +1,39 @@
+package org.keycloak.test.framework.config;
+
+import io.smallrye.config.SmallRyeConfig;
+import io.smallrye.config.SmallRyeConfigBuilder;
+import org.eclipse.microprofile.config.spi.ConfigSource;
+
+import java.util.ArrayList;
+import java.util.List;
+
+public class KeycloakSmallryeConfig {
+
+    private SmallRyeConfig config;
+
+    public KeycloakSmallryeConfig() {
+        List<ConfigSource> configSources = new ArrayList<>();
+
+        String configProperty = System.getProperty("kc.test.config");
+        if (configProperty != null) {
+            configSources.add(new FileConfigSource());
+        }
```

Comment:
Not sure if there is more elegant way to check that file based config file was provided with -Dkc.test.config